### PR TITLE
Grafana Operator Addon and Make Repository Parameter for Helm Addon Optional

### PR DESCRIPTION
**Pull request by elamaran11**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/helm-addon/helm-version-checker.ts [hunk 1]
```diff
@@ -67,14 +67,18 @@ export interface CheckVersionResult {
  */
 export function listChartVersions(chart: HelmChartVersion): string[] {
     // TODO make function async and use async HTTP client to get results
-    if(chart.repository.startsWith("oci:")){
-        return []; // skip for now if OCI repo
+    if (chart.repository) {
```
Comment (shapirov103) [inline lib/addons/helm-addon/helm-version-checker.ts:70]: I am unclear on what changed here? (vscode setting should be set to format modified lines only).

File: mkdocs.yml [hunk 2]
```diff
@@ -44,6 +44,7 @@ nav:
     - External Secrets: 'addons/external-secrets.md'
     - EMR on EKS: 'addons/emr-eks.md'
     - FluxCD: 'addons/fluxcd.md'
+    - GrafanaOperator: 'addons/grafana-operator.md'
```
Comment (shapirov103) [inline mkdocs.yml:47]: This should be a human readable name so Grafana Operator (two words)

File: docs/addons/grafana-operator.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# FluxCD Add-on
+
+The [grafana-operator](https://github.com/grafana-operator/grafana-operator#:~:text=The%20grafana%2Doperator%20is%20a,an%20easy%20and%20scalable%20way) is a Kubernetes operator built to help you manage your Grafana instances inside and outside Kubernetes. Grafana Operator makes it possible for you to manage and create Grafana dashboards, datasources etc. declaratively between multiple instances in an easy and scalable way. Using grafana-operatorit will be possible to add AWS data sources such as Amazon Managed Service for Prometheus, Amazon CloudWatch, AWS X-Ray to Amazon Managed Grafana and create Grafana dashboards on Amazon Managed Grafana from your Amazon EKS cluster. This enables us to use our Kubernetes cluster to create and manage the lifecyle of resources in Amazon Managed Grafana in a Kubernetes native way. This ultimately enables us to use GitOps mechanisms using CNCF projects such as FluxCD to create and manage the lifecyle of resources in Amazon Managed Grafana.
```
Comment (shapirov103) [inline docs/addons/grafana-operator.md:3]: "Using grafana-operatorit" = Using grafana-operator, it

Comment (shapirov103) [inline docs/addons/grafana-operator.md:3]: replace "us" with "customers" throughout.

**Review by shapirov103 — COMMENTED** [review]
a few minor comments

File: docs/addons/grafana-operator.md (hunk 3 above)
Comment (elamaran11) [reply docs/addons/grafana-operator.md]: Fixed

Comment (elamaran11) [reply docs/addons/grafana-operator.md]: Fixed

File: mkdocs.yml (hunk 2 above)
Comment (elamaran11) [reply mkdocs.yml]: Fixed

**Comment by elamaran11** [comment]
@shapirov103 All issues addressed, Can you initiate `e2e`

File: lib/addons/grafana-operator/index.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * Default props to be used when creating the Helm chart
+ */
+const defaultProps: HelmAddOnProps & GrafanaOperatorAddonProps = {
+  name: 'grafana-operator',
+  chart: 'oci://ghcr.io/grafana-operator/helm-charts/grafana-operator',
+  namespace: 'grafana-operator',
+  release: 'grafana-operator',
+  version: 'v5.0.0-rc1',
```
Comment (shapirov103) [inline lib/addons/grafana-operator/index.ts:24]: Why are we using an RC chart here instead of GA?

Comment (elamaran11) [reply lib/addons/grafana-operator/index.ts:24]: Thats their latest chart which we were asked to use by GO team. I can change it in future release if they update it. It works very well with this version.

**Review by shapirov103 — COMMENTED** [review]
LGTM

**Pull request merged**